modules/ximgproc/include/opencv2/ximgproc/scansegment.hpp

```diff
@@ -0,0 +1,127 @@
+ */
+		class CV_EXPORTS_W ScanSegment : public Algorithm
+		{
+		public:
+			/** @brief Returns the actual superpixel segmentation from the last image processed using iterate.
+			Returns zero if no image has been processed.
+			 */
+			CV_WRAP virtual int getNumberOfSuperpixels() = 0;
+
+			/** @brief Calculates the superpixel segmentation on a given image with the initialized
+			parameters in the ScanSegment object. This function can be called again for other images
+			without the need of initializing the algorithm with createScanSegment(). This save the
+			computational cost of allocating memory	for all the structures of the algorithm.
+			@param img Input image. Supported format: CV_8UC3. Image size must match with the initialized
+			image size with the function createScanSegment(). It MUST be in Lab color space.
+			 */
+			CV_WRAP virtual void iterate(InputArray img) = 0;
+
+			/** @brief Returns the segmentation labeling of the image. Each label represents a superpixel,
+			and each pixel is assigned to one superpixel label.
+			@param labels_out Return: A CV_32UC1 integer array containing the labels of the superpixel
+			segmentation. The labels are in the range [0, getNumberOfSuperpixels()].
+			*/
+			CV_WRAP virtual void getLabels(OutputArray labels_out) = 0;
+
+			/** @brief Returns the mask of the superpixel segmentation stored in the ScanSegment object.
+			@param image Return: CV_8UC1 image mask where -1 indicates that the pixel is a superpixel border,
+			and 0 otherwise.
+			@param thick_line If false, the border is only one pixel wide, otherwise all pixels at the border
+			are masked.
+			The function return the boundaries of the superpixel segmentation.
+			*/
+			CV_WRAP virtual void getLabelContourMask(OutputArray image, bool thick_line = false) = 0;
+
+			virtual ~ScanSegment() {}
+		};
+
+		/** @brief Initializes a ScanSegment object.
+		@param image_width Image width.
+		@param image_height Image height.
+		@param num_superpixels Desired number of superpixels. Note that the actual number may be smaller
+		due to restrictions (depending on the image size). Use getNumberOfSuperpixels() to
+		get the actual number.
+		@param threads Number of processing threads for parallelisation. Default -1 uses the maximum number
+		of threads. In practice, four threads is enough for smaller images and eight threads for larger ones.
+		@param merge_small merge small segments to give the desired number of superpixels. Processing is
+		much faster without merging, but many small segments will be left in the image.
+		The function initializes a ScanSegment object for the input image. It stores the parameters of
+		the image: image_width and image_height. It also sets the parameters of the F-DBSCAN superpixel
+		algorithm, which are: num_superpixels, threads, and merge_small.
+		*/
+		CV_EXPORTS_W cv::Ptr<ScanSegment> createScanSegment(int image_width, int image_height, int num_superpixels, int threads = -1, bool merge_small = true);
```

Comment:
please add
//! @}

Reply:
inserted the @addtogroup block as requested